### PR TITLE
Feature/updated ssh cert auth retargeted to master

### PR DESCRIPTION
**Pull request by sherodtaylor**
This pull request addresses #14413. It adds certificate support for the ssh provisioner. It adds a new parameter certificate which is used to authenticate ssh sessions. I updated the provisioner_test and docs to reflect this. #18017. Also updated the communicator test to validate ssh certificates. Since this was blocked by dev I decided up create this pr for dev. Reopened this as it was closed #18421

### CODE REVIEW

**Comment by sherodtaylor** [comment]
@jbardin @apparentlymart I'm back! I reopened this PR and rebased 0.12-dev

**Comment by sherodtaylor** [comment]
and updated based on the old commits

**Comment by sherodtaylor** [comment]
@apparentlymart i'll retarget this to master. Most of the newer work is on here. I left my last job at paxos.

**Comment by sherodtaylor** [comment]
@mildwonkey @apparentlymart anybody :) can i get a look at this?

**Comment by jbardin** [comment]
Hi @sherodtaylor,

Sorry about the delay here. I do intend to review this soon. Once we get master into a stable state, then I'll be going through the backlog of pending PRs.

**Comment by sherodtaylor** [comment]
@jbardin hi thanks for the response i was wondering how long until this would take?

File: website/docs/provisioners/connection.html.markdown [hunk 1]
```diff
@@ -77,6 +77,11 @@ provisioner "file" {
   [the `file` function](/docs/configuration/functions/file.html). This takes
   preference over the password if provided.
 
+* `certificate` - The contents of an signed CA Certificate. The certificate argument must be
+  used in conjunction with a `private_key`. These can
+  be loaded from a file on disk using the [`file()` interpolation
+  function](/docs/configuration/interpolation.html#file_path_).
```
[the remark this answers is not in the document]
Comment (sherodtaylor) [reply website/docs/provisioners/connection.html.markdown]: done

File: website/docs/provisioners/connection.html.markdown [hunk 2]
```diff
@@ -77,6 +77,11 @@ provisioner "file" {
   [the `file` function](/docs/configuration/functions/file.html). This takes
   preference over the password if provided.
 
+* `certificate` - The contents of an signed CA Certificate. The certificate argument must be
```
[the remark this answers is not in the document]
Comment (sherodtaylor) [reply website/docs/provisioners/connection.html.markdown]: changed

**Comment by sherodtaylor** [comment]
@jbardin hey I updated the pr

**Comment by jbardin** [comment]
Thanks @sherodtaylor!

**Comment by sherodtaylor** [comment]
@jbardin np :)

**Pull request merged**